plugin/main/src/kotlinx/benchmark/gradle/internal/KotlinxBenchmarkPluginInternalApi.kt

```diff
@@ -0,0 +1,18 @@
+package kotlinx.benchmark.gradle.internal
+
+import kotlin.RequiresOptIn.Level.WARNING
+
+/**
+ * Marks declarations that are **internal** to the Kotlinx Benchmark project.
+ *
+ * Classes or functions marked with this annotation have `public` visibility
+ * for historical or technical reasons. They are not intended for use by
+ * Kotlinx Benchmark users. They have no behaviour guarantees, and may lack clear
+ * semantics, documentation, and backward compatibility.
+ *
+ * Please remove references to internal API to supported stable references.
+ * If you still require access to these APIs, please raise an issue.
+ */
+@RequiresOptIn(level = WARNING)
```

Comment:
I would suggest adding a message to the annotation in addition or even instead to the KDoc as it's more visible and accessible even without the sources

Reply:
A message would indeed be helpful. But a lengthy one could overwhelm users. Especially if the internal API is used in multiple instances, as this would clutter the build log. I prefer a shorter message in addition to the documentation, such as "This API is internal to the Kotlinx Benchmark project and will be hidden in a future release."

Reply:
Thanks Alex, I forgot that `@RequiresOptIn` had a message option.

I used a more generic message @qurbonzoda `Internal Kotlinx Benchmark API. It may be changed in future releases without notice.` - WDYT?

Reply:
LGTM